lib/rex/proto/ldap/server.rb

```diff
@@ -56,12 +56,13 @@ def write(data)
         # @param udp [TrueClass, FalseClass] Listen on UDP socket
         # @param tcp [TrueClass, FalseClass] Listen on TCP socket
         # @param ldif [String] LDIF data
+        # @param auth_provider [Rex::Proto::LDAP::Auth] LDAP Authentication provider which processes authentication
         # @param ctx [Hash] Framework context for sockets
         # @param dblock [Proc] Handler for :dispatch_request flow control interception
         # @param sblock [Proc] Handler for :send_response flow control interception
         #
         # @return [Rex::Proto::LDAP::Server] LDAP Server object
-        def initialize(lhost = '0.0.0.0', lport = 389, udp = true, tcp = true, ldif = nil, comm = nil, ctx = {}, dblock = nil, sblock = nil)
+        def initialize(lhost = '0.0.0.0', lport = 389, udp = true, tcp = true, ldif = nil, comm = nil, auth_provider = nil, ctx = {}, dblock = nil, sblock = nil)
```

Comment:
I haven't had a chance to dive through the code fully yet to see if the existing callsites are impacted by this change; is it worth adding this new argument to the end to make sure it's a backwards compatible API change

Reply:
There is currently only one caller "the module mixin", in theory auth will be consistently extended in the rex class however this offers a place to inject a different provider. I don't see a strong reason to avoid it, especially since these are all named parameters.